src/battle_script_commands.c

```diff
@@ -1727,7 +1727,7 @@ static void AccuracyCheck(bool32 recalcDragonDarts, const u8 *nextInstr, const u
             gBattlescriptCurrInstr = nextInstr;
         else if (gStatuses3[gBattlerTarget] & (STATUS3_SEMI_INVULNERABLE))
             gBattlescriptCurrInstr = failInstr;
-        else if (!JumpIfMoveAffectedByProtect(gCurrentMove))
+        else if (!JumpIfMoveAffectedByProtect(gCurrentMove) || gProtectStructs[gBattlerTarget].maxGuarded)
```

Comment:
I'm confused why this fixes it

Reply:
I am trying to remember my thought process when I wrote it, but it's coming up blank.
I know I arrived at that point somehow, but it got lost in wrestling with tests.

Reply:
I don't fully understand it but it seems correct. Though, after the next instr there is a another Dmax check. I think there it needs a negated check for maxGuard in this case. Otherwise I assume absorbing abilities are going to be triggered even if Mon uses maxguard.

Reply:
I would probably do it like this:
```
        if (GetActiveGimmick(gBattlerAttacker) == GIMMICK_DYNAMAX)
        {
            if (gProtectStructs[gBattlerTarget].maxGuarded)
                gBattlescriptCurrInstr = nextInstr;
            else
                AbilityBattleEffects(ABILITYEFFECT_ABSORBING, gBattlerTarget, 0, 0, gCurrentMove);
        }
```
and remove check `mxGuard` check above